flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/failover/flip1/RestartPipelinedRegionFailoverStrategyTest.java

```diff
@@ -260,6 +260,41 @@ public void testRegionFailoverForMultipleVerticesRegions() throws Exception {
 				.restarts(v1, v2, v3, v4, v5, v6);
 	}
 
+	/**
+	 * Tests region failover does not restart vertexes which are already in initial CREATED state.
+	 * <pre>
+	 *     (v1) ---> (v2) --|--> (v3) ---> (v4) --|--> (v5) ---> (v6)
```

Comment:
seems overly complex; wouldn't it be sufficient to have `v2` and `v3`?

Reply:
true, although if we agree on the next @zhuzhurk's suggestion, I would leave it to include all states which do not require restart.

Reply:
I simplified the test